internal/core/src/indexbuilder/ScalarIndexCreator.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2019-2020 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License
+
+#include "indexbuilder/helper.h"
+#include "indexbuilder/ScalarIndexCreator.h"
+#include "index/IndexFactory.h"
+
+#include <string>
+
+namespace milvus::indexbuilder {
+
+ScalarIndexCreator::ScalarIndexCreator(CDataType dtype, const char* type_params, const char* index_params) {
+    dtype_ = dtype;
+    // TODO: move parse-related logic to a common interface.
+    Helper::ParseFromString(type_params_, std::string(type_params));
```

Comment:
is it safe to convert "char*" to "std::string" without length ?

Reply:
here, yes. But i have already considered using pb binary format instead of text format.